generator/internal/discovery_to_proto.cc

```diff
@@ -243,20 +245,26 @@ Status ProcessMethodRequestsAndResponses(
     std::string resource_name = CapitalizeFirstLetter(resource.first);
     auto const& methods = resource.second.json().find("methods");
     for (auto m = methods->begin(); m != methods->end(); ++m) {
-      auto response_type_name =
-          DetermineAndVerifyResponseTypeName(*m, resource.second, types);
-      if (!response_type_name) {
-        return Status(response_type_name.status().code(),
-                      response_type_name.status().message(),
+      auto response_type =
+          DetermineAndVerifyResponseType(*m, resource.second, types);
+      if (!response_type) {
+        return Status(response_type.status().code(),
+                      response_type.status().message(),
                       GCP_ERROR_INFO()
                           .WithMetadata("resource", resource.first)
                           .WithMetadata("method", m.key())
-                          .Build(response_type_name.status().code()));
+                          .Build(response_type.status().code()));
+      }
+      std::string response_type_name;
+      if (*response_type != nullptr) {
+        response_type_name = (*response_type)->name();
+        DiscoveryTypeVertex const* node = *response_type;
+        resource.second.AddResponseType(response_type_name, node);
```

Comment:
This is new. Can we verify the behavior in one of the `ProcessMethodRequestsAndResponsesTest`s?